### PR TITLE
Resets the overall pattern of defining subcommands, removes the exist…

### PR DESCRIPTION
**Pull request by mkrisher**
…ing commands and will resurrect when needed

This PR resets the overall pattern for defining subcommands to use Clap a little more. Small improvements provided by clap like incorporating the version of the Crate. Plus trapping an unknown subcommand. It traps errors now as well.

It removes the existing three commands of install, create, and get. We'll resurrect that logic as it makes sense when we introduce the commands to manage databases in the local Postgres instance.

This introduces an empty `init` command. I didn't want the PR to get too large, so I'm breaking it up. This just introduces the command, but it doesn't do anything yet. In the next PR, it will persist the configuration file, either in the default location or one specified using the --file-path argument.

```
cargo run
    Finished dev [unoptimized + debuginfo] target(s) in 0.02s
     Running `target/debug/tembo`
The CLI for Tembo

Usage: tembo [COMMAND]

Commands:
  init         Initializes a local environment or project; generates configuration
  completions  Generate shell completions for your shell to stdout
  help         Print this message or the help of the given subcommand(s)

Options:
  -h, --help     Print help
  -V, --version  Print version

For more information about a specific command, try `tembo <command> --help`
The source code for tembo is available at: https://github.com/tembo-io/tembo-cli
```

```
cargo run -- init
    Finished dev [unoptimized + debuginfo] target(s) in 0.02s
     Running `target/debug/tembo init`
Soon this will persist a configuration file
```

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -1,34 +1,67 @@
...
-impl SubCommand for SubCommands {
-    fn execute(&self) {
-        match self {
-            SubCommands::Get(cmd) => cmd.execute(),
-            SubCommands::Create(cmd) => cmd.execute(),
-            SubCommands::Install(cmd) => cmd.execute(),
-        }
+    // Check which subcommand the user ran...
+    let res = match command.get_matches().subcommand() {
+        Some(("init", sub_matches)) => cmd::init::execute(sub_matches),
```
Comment (mkrisher) [inline src/main.rs:17]: annotation: this is what adding a new command looks like now, not we trap any command that doesn't match on line 32

File: src/main.rs [hunk 2]
```diff
@@ -1,34 +1,67 @@
...
+                &mut std::io::stdout().lock(),
+            );
+            Ok(())
+        })(),
+        _ => unreachable!(),
+    };
+
+    if let Err(_) = res {
+        // TODO: adding logging, log error
+        std::process::exit(101);
```
Comment (mkrisher) [inline src/main.rs:37]: annotation: we're now trapping any errors that bubble up from commands, eventually adding some logging to spit out more information, we return a proper process exit code

File: src/main.rs [hunk 3]
```diff
@@ -1,34 +1,67 @@
...
+        .about(crate_description!())
+        .author("Tembo <ry@tembo.io>")
+        .version(VERSION)
+        .propagate_version(true)
+        .arg_required_else_help(true)
+        .after_help(
+            "For more information about a specific command, try `tembo <command> --help`\n\
+             The source code for tembo is available at: https://github.com/tembo-io/tembo-cli",
+        )
+        .subcommand(cmd::init::make_subcommand())
```
Comment (mkrisher) [inline src/main.rs:53]: annotation: this lets us list the subcommands with some additional info if we want

File: src/cmd/init.rs [hunk 4]
```diff
@@ -0,0 +1,37 @@
+use clap::builder::NonEmptyStringValueParser;
+use clap::{Arg, ArgAction, ArgMatches, Command};
+use std::error::Error;
+
+// Create clap subcommand arguments
+pub fn make_subcommand() -> Command {
+    Command::new("init")
+        .about("Initializes a local environment or project; generates configuration")
+        .arg(
+            Arg::new("dry-run")
```
Comment (mkrisher) [inline src/cmd/init.rs:10]: annotation: dry-run will just spit out a message to stdout letting the user know where the configuration file would be persisted

**Review by ChuckHend — APPROVED** [review]

**Pull request merged**